contracts/MyStrategy.sol

```diff
@@ -100,26 +122,47 @@ contract MyStrategy is BaseStrategy, ReentrancyGuardUpgradeable {
         LOCKER.delegate(delegate);
     }
 
-    ///@dev Should we check if the amount requested is more than what we can return on withdrawal?
+    /// @dev Should we check if the amount requested is more than what we can return on withdrawal?
     function setWithdrawalSafetyCheck(bool newWithdrawalSafetyCheck) external {
         _onlyGovernance();
         withdrawalSafetyCheck = newWithdrawalSafetyCheck;
     }
 
-    ///@dev Should we processExpiredLocks during reinvest?
+    /// @dev Should we processExpiredLocks during reinvest?
     function setProcessLocksOnReinvest(bool newProcessLocksOnReinvest) external {
         _onlyGovernance();
         processLocksOnReinvest = newProcessLocksOnReinvest;
     }
 
-     ///@dev Change the contract that handles bribes
+     /// @dev Change the contract that handles bribes
     function setBribesProcessor(IBribesProcessor newBribesProcessor) external {
         _onlyGovernance();
         bribesProcessor = newBribesProcessor;
     }
 
+     /// @dev Sets the redirection path for a given token as well as the redirection fee to 
+     ///      process for this recepient.
+     /// @notice There can only be one recepient per token, calling this function for the same
+     /// @notice token will replace the previous one.
+     /// @notice Adding a token to this mapping means that the full amount (minus the fee) of this
+     /// @notice token, claimed from HiddenHands, will be transfer to this recepient.
+     /// @param token Bribe token to redirect
+     /// @param recepient Address where redirected token will be transferred
+     /// @param redirectionFee Fee to be processed for the redirection service, different per recepient
+    function setRedirectionToken(address token, address recepient, uint256 redirectionFee) external {
+        _onlyGovernance();
+        require(token != address(0), "Invalid token address");
+        require(recepient != address(0), "Invalid recepient address");
+        require(redirectionFee <= MAX_BPS, "Invalid redirection fee");
+
+        // Sets redirection path for a given token
+        bribesRedirectionPaths[token] = recepient;
+        // Sets redirection fees for a given recepient
+        redirectionFees[recepient] = redirectionFee;
```

Comment:
Why's the fee based on recipient and not tokens? Also, might be better to have a separate function for setting redirection fees if they are based only on recipients

Reply:
That's a good question, will clarify with @Tritium-VLK if there could be multiple tokens for one recipient and, if this is the case, if the fees would be different per token.

Reply:
Mapped fees to the tokens as well, made more sense indeed.